airflow/www/views.py

```diff
@@ -1735,6 +1747,44 @@ def gantt(self):
             root=root,
         )
 
+    @expose('/redirect')
+    @login_required
+    @wwwutils.action_logging
+    def redirect(self):
+
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        redirect_to = request.args.get('redirect_to')
```

Comment:
Unvalidated redirect_to params is a security concern:
https://www.owasp.org/index.php/Unvalidated_Redirects_and_Forwards_Cheat_Sheet

Reply:
@mistercrunch this `redirect_to` variable gets passed to operator's `get_redirect_url` method, which takes the decision and not blindly redirects to the given value. Is this also a security concern? And should we rename the param name?

Reply:
I think the concern is for phishing attacks where you'd get a legit looking email with a link to a domain you recognize (`airflow.mycompany.com?foo=bar&redirect_to=airflow.almostmycompany.com`) with a prompt to login. 

Regardless of where the `redirect_to` variable comes from, the redirect endpoint is exposed and abusable.

I think that the guideline is to ensure that the `redirect_to` is on the same domain

Reply:
I think having whitelisted domains by the admin (airflow.cfg) is something here. By default it should be None or “origin”. I don’t trust operators as they are decoupled from the web server.

Reply:
And the check should happen in the web server not in the operator.